engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/RenderGraphOverlay.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.ingame.metrics;
+
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.input.MouseInput;
+import org.terasology.registry.In;
+import org.terasology.rendering.dag.Node;
+import org.terasology.rendering.dag.RenderGraph;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.itemRendering.StringTextRenderer;
+import org.terasology.rendering.nui.widgets.UITreeView;
+import org.terasology.rendering.nui.widgets.treeView.GenericTree;
+import org.terasology.rendering.world.WorldRenderer;
+
+/**
+ * Displays the content of the WorldRenderer renderGraph.
+ */
+public class RenderGraphOverlay extends CoreScreenLayer {
+    private static final Logger logger = LoggerFactory.getLogger(RenderGraphOverlay.class);
+
+    @In
+    private WorldRenderer worldRenderer;
+    
+    private UITreeView<RenderNodeInfo> renderGraphTreeView;
+
+    @Override
+    public void initialise() {
+        WidgetUtil.trySubscribe(this, "close", widget -> getManager().closeScreen(RenderGraphOverlay.this));
+        renderGraphTreeView = find("renderGraphTreeView", UITreeView.class);
+        renderGraphTreeView.subscribeNodeClick((event, treeNode) -> {
+            if (event.getMouseButton() == MouseInput.MOUSE_LEFT) {
+                RenderNodeInfo renderNodeInfo = (RenderNodeInfo)treeNode.getValue();
+                Node renderNode = renderNodeInfo.getNode();
+                String uriString = renderNode.getUri().toString();
+                logger.info(uriString);
```

Comment:
Is there a particular reason you're first fetching the node and then calling `toUri` on it, instead of `String uriString = renderNodeInfo.toString(); ` (which also takes care of the `node == null` case?)

Reply:
I wasn't sure exactly what I wanted to happen on mouse-click yet.   I was kind of thinking a pop-up panel with more information about the node, but as a placeholder to detect the click and make sure it was working right, I decided to dump something to the console.    

The fact that renderNodeInfo.toString() currently shows the url or ROOT isn't behavior I planned to keep.   toString() traditionally dumps the contents of an object in a format that the developer can use for debugging, but isn't recommended for actual program logic, so I'll very likely change it later to also dump input and output information as well as the uri.   I really hate the root node = null situation.   I'd like to do away with it entirely.

But the answer for this particular section of code is that it's all going to be deleted "soon".